lib/http/chainable.rb

```diff
@@ -76,30 +76,28 @@ def request(verb, uri, options = {}) # rubocop:disable Style/OptionHash
       branch(options).request verb, uri
     end
 
+    TIMEOUT_USAGE = "Use `.timeout(global_timeout_in_seconds)` or `.timeout(connect: x, write: y, read: z)`."
+
     # @overload timeout(options = {})
-    #   Syntax sugar for `timeout(:per_operation, options)`
-    # @overload timeout(klass, options = {})
-    #   Adds a timeout to the request.
-    #   @param [#to_sym] klass
-    #     either :null, :global, or :per_operation
+    #   Adds per operation timeouts to the request
     #   @param [Hash] options
     #   @option options [Float] :read Read timeout
     #   @option options [Float] :write Write timeout
     #   @option options [Float] :connect Connect timeout
-    def timeout(klass, options = {}) # rubocop:disable Style/OptionHash
-      if klass.is_a? Hash
-        options = klass
-        klass   = :per_operation
-      end
-
-      klass = case klass.to_sym
-              when :null          then HTTP::Timeout::Null
-              when :global        then HTTP::Timeout::Global
-              when :per_operation then HTTP::Timeout::PerOperation
-              else raise ArgumentError, "Unsupported Timeout class: #{klass}"
-              end
-
-      %i[read write connect].each do |k|
+    # @overload timeout(global_timeout)
+    #   Adds a global timeout to the full request
+    #   @param [Numeric] global_timeout
+    def timeout(options, extra_arg = nil)
+      raise ArgumentError, TIMEOUT_USAGE if extra_arg
```

Comment:
This is gonna be a breaking change, so I think it's fine to just change method signature to accept single argument only.

Reply:
I debated this, but thought a bit of effort to ease the transition was worth it. I'll remove it. No worries.

Reply:
Per #447 we might end up redoing the whole thing with keyword arguments anyway, but don't worry about that for this PR